src/components/MasterDetailNav/MasterDetailNav.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import View from '../View';
+import { WithQueries } from 'avenger/lib/react';
+import { currentView } from '../../queries';
+import { doUpdateCurrentView } from '../../commands';
+import { constNull } from 'fp-ts/lib/function';
+import { CurrentView } from 'src/model';
+
+import './master-detail-nav.scss';
+
+type OptionType = { value: CurrentView; label: string };
+
+const master: OptionType = { value: 'search', label: 'Search' }
+const detail: OptionType = { value: 'detail', label: 'Detail' }
+
+export default class MasterDetailNav extends React.Component {
+  goToMaster = () => {
+    doUpdateCurrentView(master.value).run();
+  };
+
+  render() {
+    return <WithQueries queries={{currentView}} render={queries => {
+      const isDetails = queries.fold(constNull, constNull, q => q.currentView == 'detail');
```

Comment:
minor, and doesn't change the functionality, but usually when "folding" it is advisable to have a single target type, which in this case should be `boolean` (and not `boolean | null`).
```suggestion
      const isDetails = queries.fold(constFalse, constFalse, q => q.currentView == 'detail');
```